File: backend/backend/src/main/java/com/backend/backend/dto/MatchSummaryResponse.java
package com.backend.backend.dto;

import java.time.Instant;
import java.util.List;

import com.backend.backend.model.MatchStatus;

public record MatchSummaryResponse(String id,
                                   String userId,
                                   Instant createdAt,
                                   Instant processedAt,
                                   MatchStatus status,
                                   Integer durationSeconds,
                                   String videoPath,
                                   String originalFilename,
                                   String player1Name,
                                   String player2Name,
                                   String matchTitle,
                                   String thumbnailPath,
                                   MatchStatisticsResponse statistics,
                                   List<ShotResponse> shots,
                                   List<EventResponse> events,
                                   HighlightsResponse highlights) {
}

File: Frontend/app/src/main/java/com/example/myapplication/data/api/dto/MatchSummaryResponse.kt
package com.example.myapplication.data.api.dto

/**
 * Response from GET /api/matches
 */
data class MatchSummaryResponse(
    val id: String,
    val createdAt: String,
    val processedAt: String?,
    val status: String,
    val durationSeconds: Int?,
    val originalFilename: String?
)
